sound/soc/sdw_utils/soc_sdw_utils.c

```diff
@@ -1204,6 +1228,53 @@ int asoc_sdw_parse_sdw_endpoints(struct snd_soc_card *card,
 				    !(dai_info->quirk_exclude ^ !!(dai_info->quirk & ctx->mc_quirk)))
 					continue;
 
+				dlc.dai_name = dai_info->dai_name;
+				codec_dai = snd_soc_find_dai_with_mutex(&dlc);
+				if (!codec_dai) {
+					dev_err(dev, "codec dai %s not registered yet\n",
+						dlc.dai_name);
+					return -EPROBE_DEFER;
+				}
+
+				/* No need to check SDCA functions if there is only 1 endpoint present */
+				if (adr_dev->num_endpoints == 1)
+					goto skip_sdca_function_check;
+
+				sdw_codec_name = _asoc_sdw_get_codec_name(dev, codec_info, adr_link, i);
+				if (!sdw_codec_name)
+					return -ENOMEM;
+
+				sdw_dev = bus_find_device_by_name(&sdw_bus_type, NULL, sdw_codec_name);
+				if (!sdw_dev) {
+					dev_err(dev, "codec %s not found\n", sdw_codec_name);
+					return -EINVAL;
+				}
+
+				slave = dev_to_sdw_dev(sdw_dev);
+				if (!slave)
+					return -EINVAL;
+
+				/* Mack sure BIOS provides SDCA properties */
+				if (!slave->sdca_data.interface_revision)
+					goto skip_sdca_function_check;
+
+				for (k = 0; k < slave->sdca_data.num_functions; k++) {
+					int dai_type = asoc_sdw_get_dai_type(
+							slave->sdca_data.function[k].type);
+					dev_dbg(&slave->dev, "function: %s dai_type %d\n",
+						slave->sdca_data.function[k].name, dai_type);
+					if (dai_type == dai_info->dai_type) {
+						dev_dbg(&slave->dev, "DAI type %d found\n",
+							dai_type);
+						break;
+					}
+				}
+				if (k == slave->sdca_data.num_functions) {
+					dev_dbg(&slave->dev, "DAI type %d not found, skip endpoint\n",
```

Comment:
Also here, maybe  "SDCA device function for DAI type %d not supported, skip endpoint"